main.go

```diff
@@ -818,6 +818,104 @@ The validate command expects a configuration file as its only argument. Local fi
 			return action.ListBootEntries(cfg)
 		},
 	},
+	{
+		Name:  "grubedit",
+		Usage: "grubedit [set] KEY=VALUE",
+		Description: "Show or edit grub variables.\n\n" +
+			"Use `grubedit set KEY=VALUE` to set a grub variable. If the value is empty, it will remove the variable.\n\n" +
+			"Use `grubedit` to list all the grub variables found in the grub environment file.",
+		Action: func(c *cli.Context) error {
+			config, err := agentConfig.Scan(collector.Directories(constants.GetUserConfigDirs()...), collector.NoLogs, collector.StrictValidation(c.Bool("strict-validation")))
+			if err != nil {
+				return err
+			}
+			grubEnv := c.String("grub-file")
+			if _, err := config.Fs.Stat(grubEnv); err != nil && os.IsNotExist(err) {
+				// If file doesn't exist, we just log it and return
+				// Is not really an error, its just that there is no file so no vars are set
+				config.Logger.Logger.Info().Msgf("No file %s found", grubEnv)
+				return nil
+			}
+			vars, err := utils.ReadPersistentVariables(grubEnv, config)
+			if err != nil {
+				config.Logger.Logger.Error().Err(err).Msgf("Error reading grub variables from %s", grubEnv)
+				return err
+			}
+			if len(vars) == 0 {
+				config.Logger.Logger.Info().Msgf("No grub variables found in %s", grubEnv)
+			} else {
+				// Format the variables into a nice aligned list
+				maxKeyLen := 0
+				for k := range vars {
+					if len(k) > maxKeyLen {
+						maxKeyLen = len(k)
+					}
+				}
+				var b strings.Builder
+				for k, v := range vars {
+					b.WriteString(fmt.Sprintf("%-*s = %s\n", maxKeyLen, k, v))
+				}
+				config.Logger.Logger.Info().Msgf("Grub variables found in %s:\n%s", grubEnv, b.String())
+			}
+			return nil
+		},
+		Subcommands: []*cli.Command{
+			{
+				Name:      "set",
+				Usage:     "Add or Edit a grub variable. Setting an existing variable to an empty value will remove it",
+				UsageText: "grubedit set KEY=VALUE",
+				Before: func(c *cli.Context) error {
+					if c.Args().Len() != 1 {
+						cli.HelpPrinter(c.App.Writer, "File and key=value arguments missing\n\n", c.Command)
+						_ = cli.ShowSubcommandHelp(c)
+						return fmt.Errorf("")
+					}
+					return nil
+				},
+				Action: func(c *cli.Context) error {
+					config, err := agentConfig.Scan(collector.Directories(constants.GetUserConfigDirs()...), collector.NoLogs, collector.StrictValidation(c.Bool("strict-validation")))
+					if err != nil {
+						return err
+					}
+
+					// Transform the key=value argument into a map
+					parts := strings.SplitN(c.Args().First(), "=", 2)
```

Comment:
Where is the [padding](https://github.com/kairos-io/kairos-agent/pull/846/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R856) removed?

Reply:
Im not sure what does that means. The padding you linked is only for listing the grub variables for the user, for show basically so they look nice, but we dont add any padding to them when we manage them.